packages/eas-cli/src/commands/env/unlink.ts

```diff
@@ -0,0 +1,96 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASEnvironmentFlag, EASNonInteractiveFlag } from '../../commandUtils/flags';
+import { EnvironmentVariableMutation } from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { getDisplayNameForProjectIdAsync } from '../../project/projectUtils';
+import { selectAsync } from '../../prompts';
+import { promptVariableEnvironmentAsync } from '../../utils/prompts';
+
+export default class EnvironmentVariableUnlink extends EasCommand {
+  static override description = 'link a shared environment variable to the current project';
+
+  static override hidden = true;
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable',
+    }),
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: { name, 'non-interactive': nonInteractive, environment },
+    } = await this.parse(EnvironmentVariableUnlink);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentVariableUnlink, {
+      nonInteractive,
+    });
+
+    if (!environment) {
+      environment = await promptVariableEnvironmentAsync(nonInteractive);
+    }
+
+    const projectDisplayName = await getDisplayNameForProjectIdAsync(graphqlClient, projectId);
+
+    const { appVariables, sharedVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
+      graphqlClient,
+      projectId,
+      environment
+    );
+    const sharedVariablesIds = sharedVariables.map(v => v.id);
+    const linkedVariables = appVariables.filter(({ id }) => sharedVariablesIds.includes(id));
```

Comment:
Would it be possible to do sth like:
```suggestion
    const { appVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
      graphqlClient,
      projectId,
      environment
    );
    const linkedVariables = appVariables.filter(({ scope }) => scope === EnvironmentVariableScope.Shared);
```
?

This way we could only do 1 query instead of 2 (assuming that we would also modify queries).